### DISCUSSION TITLE
路由跳转时传的参数，刷新页面参数仍然生效的问题

### DISCUSSION THREAD

**Discussion by xclw2000** [umijs/umi#8461] [Q&A]
我通过如果下方式在从A页面跳转到B页面时传参数(state)
```
          <Link
            to={{
              pathname: `/B`,
              state: {
                schoolTypeOptions,
                selectedSchoolType,
                schoolYearOptions,
                selectedSchoolYear,
                categoryTreeData,
                selectedCategory,
                selectedCategoryKeys,
                expandedCategoryKeys,
              },
            }}
          >
            详情
          </Link>
```
B页面还可以跳转回A页面，跳转时再把参数(state)传回来，以便我可以把A页面恢复到跳转前的状态。
现在的问题时，跳转回来之后，如果点浏览器的刷新按钮，这个参数还在，我希望在点刷新时，以没有参数的方式加载页面。我想到的办法时当在A页面中发现有参数时，用完这些参数，能把它们清除，即只使用一次。如何实现？
A---参数--->B---参数--->A---刷新--->A(此时state还在)
更进一步，这些参数保存在哪里了，我都已经刷新页面了，相当于整个应用程序都重新加载了，可是它们还在，我看localStorage中也没有这些数据。

**Comment by fz6m** [comment]
给个复现 demo 看看。

**Reply by xclw2000** [reply]
[history_state_demo.zip](https://github.com/umijs/umi/files/9068248/history_state_demo.zip)
这是复现的代码
1. 刚进来是数量是0
2. 点"+"号增加数量
3.点"Go To Page B" 进入新页面
4.在新页面中点”Go To Page A“，返回之前的页面
5.页面的数量显示的是之前的数量
6.点浏览器的刷新，数量没有变成0，还是之前的数量

**Comment by lbrjms** [comment]
你好 最后怎么实现了
